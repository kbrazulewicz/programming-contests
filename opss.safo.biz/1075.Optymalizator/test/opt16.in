10
500 -150961
-323 -451 -309 193 -323 255 -190 31 -368 148 83 -442 344 394 -153 -192 391 119 334 -50 50 339 444 -151 -449 -127 287 -134 -413 85 -95 -236 487 96 457 16 -149 119 -453 336 -233 130 394 -389 376 93 272 -381 65 106 69 -33 297 365 -332 348 -262 455 -286 -174 393 119 442 380 67 400 -104 -230 19 295 -394 -361 -75 -147 102 153 446 226 272 363 -168 193 -170 -19 58 -2 -171 -352 306 395 326 199 14 269 431 -67 -479 179 204 40 -26 -338 -469 -249 15 134 404 -38 -140 176 177 -455 221 -492 26 -221 358 -292 279 164 -397 106 215 118 227 146 -97 248 -175 -393 -212 -349 122 -180 -98 -363 -194 -194 451 166 -166 129 -437 -445 489 442 186 347 -350 -34 364 -247 -76 79 -277 -349 -274 127 399 403 -414 39 -445 208 -289 309 346 17 -384 -351 36 302 278 99 -290 267 393 -252 467 -457 214 331 -203 -10 -90 20 141 488 -353 40 -256 86 -68 151 -206 143 -40 -8 -339 428 494 -451 -417 -376 -500 -207 -256 394 41 -289 437 108 -106 -266 -402 304 -393 -408 -355 106 484 241 -308 -84 -108 339 -440 -296 183 73 -367 177 122 68 302 122 361 -454 16 254 -243 306 -138 151 -460 313 307 -1 405 -196 -394 389 -103 -202 -194 -359 -11 218 -154 -327 -209 331 350 413 -101 -496 35 112 -450 -96 -134 159 -290 -419 162 102 394 322 -398 -349 -22 60 -460 376 358 198 -483 200 -84 215 373 207 46 -425 472 445 -420 360 57 -18 -236 -224 -358 326 -143 -344 -72 103 -170 382 254 309 -58 -354 -463 153 -303 -446 -147 465 270 -274 -475 168 153 497 466 233 357 375 -284 -379 -349 210 -201 360 -282 -420 -37 49 462 69 -142 257 -432 -105 -90 -235 301 -385 -270 423 193 -245 92 -154 105 58 -420 -38 -67 148 436 85 210 -265 297 428 -185 -239 -23 130 330 -313 -113 398 434 -351 15 236 -236 -402 159 457 205 103 -197 -190 -339 235 -375 447 235 -439 384 -55 148 181 226 -184 -206 -297 446 -23 243 185 227 177 -314 -257 -87 302 -159 -75 259 -102 -120 -86 209 394 2 334 341 89 -253 -423 -465 395 258 261 -289 -95 -184 9 382 -441 46 109 237 -416 352 2 386 45 -221 497 444 160 264 -495 406 118 339 -401 -293 -62 -324 -258 -167 -214 -145 397 191 172 258 73 -417 157 -317 -328 241 -113 -473 480 433 -194 477 -271 -182 -407 -266 -276 211 -75 -177 419 363 351 -339 48 -362 369 297 329 393 56 255 476 65 438 
500 -24851
-194 -175 28 138 110 334 116 -309 5 209 277 -419 273 -298 257 192 -83 108 -295 -182 -402 -426 -385 428 467 -477 -465 -204 440 473 297 99 150 -175 -263 113 11 -295 304 -484 415 82 -50 188 -364 -293 232 406 -333 437 -276 -234 -488 191 46 -169 -433 81 479 -493 406 276 106 408 -47 344 21 465 -451 178 -167 -184 112 283 -144 -400 -158 -412 -494 10 -122 82 -372 242 126 174 -427 193 -245 -95 -448 13 -319 -489 -79 487 355 -205 452 -244 473 137 425 437 273 133 -463 -385 -426 -104 477 -48 -170 -395 -306 -192 279 119 -147 -114 -476 -94 399 58 -83 -179 45 124 468 349 -268 441 -14 9 -270 111 -357 119 227 -283 -133 204 -479 198 -338 -285 -142 -207 -314 212 180 -289 -382 431 -379 387 104 166 363 72 367 -53 365 205 457 -405 -183 -48 215 -456 -479 -418 -400 -458 -368 -238 109 -9 408 295 -297 88 -142 173 371 -21 412 -24 -3 -373 -100 216 74 -234 -78 31 -139 239 -165 -72 283 -144 363 -265 -102 -153 350 -141 338 258 7 -107 198 -283 -82 69 -451 182 -103 46 309 298 -237 -116 -436 185 -233 277 276 103 -294 59 459 -79 -354 210 268 496 -431 459 -394 76 -148 304 294 -377 -126 343 305 271 241 115 69 -144 499 485 -107 118 263 -331 -279 -179 -420 33 242 -273 -257 10 75 164 -31 182 93 -326 -14 -113 -203 -288 82 -46 484 175 -431 405 -468 -80 391 277 -461 -494 447 112 -173 -473 -355 -431 106 240 431 -318 -95 -99 364 498 427 202 237 -276 415 171 178 -249 198 -400 -344 -418 20 -101 360 411 -95 307 376 -416 -314 -479 5 293 -386 -63 475 19 -310 -309 369 -383 393 458 193 308 481 223 -441 179 323 68 262 196 -181 122 459 -423 281 335 -339 467 -291 167 -388 -177 -396 -61 -306 -206 130 -85 263 376 -127 -44 184 -146 -321 -404 -467 -145 -484 -353 -449 -165 269 -490 -88 50 -302 426 370 -93 93 482 230 -451 -78 -224 195 -448 191 -42 428 -436 414 465 270 -55 -87 -345 300 -71 -197 -297 264 -428 66 29 475 264 -45 345 -329 -100 327 -247 301 -251 29 -4 154 -280 -194 82 136 72 399 258 -131 -188 413 -330 -407 -284 225 358 141 -209 239 116 -445 46 313 78 -202 140 331 -401 -258 -140 447 396 432 105 330 420 -323 230 178 46 -106 444 -432 -12 -340 294 198 -347 -415 -63 -379 -7 -165 434 -429 133 427 403 232 -331 115 -469 -435 400 136 -253 320 -335 477 351 211 224 295 -368 
500 -81788
-193 -222 -238 -39 215 199 434 -292 -114 -279 132 -481 -352 387 -397 169 354 -366 234 254 270 -167 427 435 -189 278 498 -113 73 -18 99 380 260 361 193 328 412 128 -112 -202 349 -480 169 -151 407 272 -482 -386 406 -396 368 -472 -63 295 -185 248 425 165 -13 -2 -352 -414 231 -240 -201 -76 -412 63 -96 -23 361 -247 349 382 102 -391 154 120 -277 412 224 -409 -208 -338 -261 107 -238 -336 -227 250 -485 273 -312 246 -467 340 -330 -26 -97 75 451 117 -320 -348 -1 -365 -239 6 255 -16 -82 332 428 -437 -154 -333 170 460 183 -205 -438 198 -432 103 -56 -46 -57 467 428 346 394 -269 315 -426 235 -185 -291 497 321 -183 333 91 1 261 154 347 -220 177 -341 464 -176 74 -486 -107 -323 -41 -301 472 278 -21 -330 172 -438 486 246 297 153 -192 294 326 125 128 -83 -374 241 -76 -175 -478 -47 336 486 277 -90 -500 22 439 311 221 411 89 -300 -418 113 114 -432 -140 -88 221 -480 58 47 145 -314 -184 -377 428 -408 300 -198 45 488 140 -177 -101 492 -303 -162 304 419 -250 245 -29 184 -141 -414 252 -429 -2 325 -409 -444 224 -412 95 -460 63 375 -367 215 -323 -470 203 -331 205 -398 -339 403 293 465 322 -105 -437 145 -421 -78 231 331 -7 81 -492 436 490 232 -476 437 272 439 -336 -243 154 193 140 -143 362 -155 -40 -477 -252 105 -159 -78 -500 -96 -80 -421 326 3 -238 171 -415 -230 107 -73 -498 483 216 -374 422 380 -116 76 73 -124 433 -65 221 245 310 -178 -150 -349 244 202 55 -336 281 -119 168 -457 -96 -395 165 -489 384 167 494 100 294 -232 332 30 344 -95 406 -370 -308 479 375 -498 -199 78 -346 -102 -220 -439 62 -438 -57 82 -43 -301 -461 -377 -289 -225 -358 57 375 -212 326 207 318 -478 -36 -424 -348 156 -444 -472 -489 209 458 17 -393 -262 78 -478 152 -127 456 -390 73 496 85 -364 -229 227 193 -1 -484 371 58 -314 394 -125 -237 -102 -469 171 -74 -106 -120 -116 411 340 475 -10 362 127 -285 318 237 288 -334 -178 424 290 -98 470 -211 -82 -307 -301 104 -61 74 -281 338 458 390 116 -148 123 -147 116 -37 -172 -42 177 455 173 -5 -455 -38 -486 219 238 304 -379 208 -55 391 -246 144 -4 193 71 -433 31 29 -190 -352 -267 433 1 349 248 329 159 -75 -364 -167 -228 -319 -353 -214 -247 385 -410 -126 446 387 -234 -300 32 -386 393 -397 -319 277 484 -9 425 69 276 278 -81 376 459 -422 
500 26301
-405 263 426 129 410 -288 382 -352 155 108 -54 42 -274 498 -74 -160 -257 381 374 -480 -283 -135 -203 287 -6 -425 -294 -130 -466 136 -476 -370 252 450 -389 162 14 493 -338 21 453 108 -436 -320 458 342 20 -298 224 246 -278 441 -36 20 228 458 447 -214 180 482 -225 204 -36 -473 -494 75 -459 21 420 -296 -106 373 -336 -42 -95 -377 153 -222 177 -123 -476 399 170 -12 271 251 298 219 -111 -21 -447 164 -317 17 -457 42 -56 -415 -85 216 141 309 441 -343 120 -153 132 -227 125 -191 -498 -499 61 -476 -10 -168 275 -212 -97 17 267 308 -319 -197 177 77 -303 -379 -486 112 -311 155 273 -369 312 -107 330 -55 -482 307 -394 -480 308 167 -103 -202 -500 -328 439 255 -459 206 416 -277 -139 93 -348 58 -433 -334 22 -244 321 -204 239 485 -459 69 282 -440 -124 389 -68 -315 56 329 335 408 -146 274 -484 -105 333 432 470 -306 377 -26 -395 444 140 127 -299 313 -225 -208 299 -183 362 433 229 90 322 13 275 231 195 463 491 -99 237 -493 296 70 -209 -381 -383 -331 445 -278 -387 86 -299 166 -101 -23 459 50 -354 -327 -16 -273 -385 158 240 391 241 -65 -294 233 336 443 -408 485 366 -116 -44 483 405 -99 -295 -130 487 -94 -463 239 -265 -152 -211 -119 373 -375 460 488 284 201 -269 377 -364 -63 462 -175 233 -445 -190 99 291 -382 82 196 19 -361 -434 359 397 -45 98 -367 -345 387 366 -472 -135 327 -131 1 28 100 -122 -484 -110 -159 -159 -377 -104 -497 -426 39 -379 156 87 141 295 153 352 192 -39 -50 -323 116 189 -456 497 -446 -277 -134 55 -397 -182 434 -29 -440 127 313 -317 -125 -332 -243 414 142 -235 1 135 -88 6 -13 -395 -181 437 134 436 126 -470 -215 -467 -247 151 88 208 -179 374 -320 -118 353 -155 417 -272 13 -325 -6 -345 292 495 290 -295 354 -223 162 -327 66 -204 -39 45 179 246 78 432 -103 18 141 -429 245 -327 -47 98 18 -130 -321 -469 397 173 39 190 -479 181 -253 375 -189 409 -100 377 57 362 274 -264 108 204 21 -142 -277 14 281 -180 187 -266 -230 -295 456 -51 88 354 -377 -21 396 -4 -339 -5 223 324 -96 -377 201 -187 -163 328 50 -202 32 -77 -492 -393 437 289 279 476 -125 -450 -319 -169 -1 -379 37 474 101 285 -30 -386 -220 45 438 -464 169 491 -150 -142 319 -248 156 204 175 164 311 464 305 91 440 -320 493 -379 364 -8 94 -99 319 -305 39 141 161 319 -313 
500 72599
356 -292 91 58 418 262 310 75 -34 -15 91 130 -51 249 73 -111 429 66 362 -355 -90 -44 399 81 -496 -62 -277 -483 109 262 117 -35 -178 60 -125 -260 174 -315 -333 -7 -478 259 -25 323 360 -452 64 -359 466 -74 -213 -124 -265 -314 -190 91 476 33 108 85 147 77 -97 -179 -11 278 61 -336 464 229 -491 -162 340 -16 -338 200 32 226 341 -2 5 -372 -274 -408 166 36 -465 494 -79 495 80 -80 73 483 241 -86 -387 155 430 -423 -116 439 268 76 -77 430 -224 455 -492 -31 305 13 450 -116 457 116 -228 492 111 -454 -160 43 318 -235 378 60 179 491 -285 110 -79 451 49 -311 27 473 -29 -345 -220 -21 124 -62 -155 74 -326 -198 43 298 147 6 344 487 -99 -485 -248 -221 75 -217 -230 -358 393 191 -55 295 -268 -176 -380 203 331 252 35 -45 -310 380 -118 216 34 -223 15 -319 -365 211 20 36 226 272 315 153 -444 -63 -353 -199 -371 92 -404 213 -232 -284 269 -401 -179 -196 407 11 -316 -211 80 218 418 -405 -248 -447 -342 272 89 385 397 256 38 -195 -307 -462 106 174 130 -445 -112 -249 123 -343 -150 444 -39 -243 308 497 398 -112 215 316 335 467 369 493 -408 310 -270 489 66 -379 -206 260 -341 -248 286 -359 159 -326 -108 -217 183 243 79 -4 -148 -113 -7 -249 -373 -291 -81 462 28 141 -192 120 -49 390 461 -130 -489 -245 -18 22 360 -232 164 19 295 -92 302 -22 3 -118 -173 356 -379 320 -41 -251 -119 378 63 -238 19 -277 382 -177 -386 -304 -455 -23 -49 27 -500 163 -353 -484 -317 442 276 -163 273 -220 -429 -400 -12 -307 -228 447 294 154 -323 -143 -232 -451 81 2 -128 -453 198 -83 24 2 -204 376 -335 -57 392 200 238 169 -462 363 -199 -39 -185 289 -494 87 -412 300 93 117 10 361 166 -57 -136 -110 -10 -438 307 -486 64 -45 391 -418 -249 -365 282 489 -344 -328 -296 -43 134 19 98 492 -42 -462 293 -448 156 155 -235 174 -402 129 -435 440 -456 -276 -194 460 180 49 -458 431 185 325 -228 341 -151 -24 151 -165 495 -251 -172 -47 140 -27 357 -204 -372 -377 -178 78 252 -113 18 -352 112 324 -391 -208 226 -349 75 -89 -172 347 104 178 -177 -245 -135 170 357 193 -25 349 -334 -167 -355 146 -44 -33 -276 60 -293 -406 209 319 -229 318 -37 497 -179 -110 260 150 -263 -136 180 412 472 -455 434 -319 239 -91 -470 -243 242 27 404 50 -154 -20 -389 53 75 -328 224 -302 490 39 
500 -11805
163 429 455 -187 18 171 -155 -70 143 -109 216 324 -18 -374 -294 -409 -280 233 495 -377 -420 328 -266 485 -97 258 210 101 248 -251 -352 -89 -321 -397 -423 -451 274 -78 332 270 -335 48 94 499 -474 301 -409 -253 386 -62 -130 466 -234 456 452 -479 214 -486 122 314 -237 -378 77 294 77 -494 344 -148 281 -472 -26 446 76 -432 -202 103 -279 -111 -298 -40 327 72 -74 -54 380 230 -33 446 -404 -58 260 212 64 189 6 -358 -452 202 346 329 230 -180 -373 159 240 -75 -238 -186 -334 -36 274 -154 -112 52 292 -232 -217 -241 66 231 201 178 -57 118 -133 450 260 267 -496 458 96 -413 -222 224 246 -482 1 -140 184 168 324 -42 -486 64 -489 -342 -316 146 -83 250 -123 -29 -72 321 -411 -353 -377 -299 415 -21 -341 363 418 -63 439 164 307 -59 -124 -8 -391 52 -198 -25 -384 -187 133 152 -189 -98 -98 -459 373 -318 -286 -186 182 189 15 449 -332 -474 -336 -413 315 -396 -397 475 45 332 -33 -494 -116 -379 333 353 287 318 5 -402 -280 -240 -9 -406 294 205 260 328 -106 128 277 415 154 442 -146 -30 -102 309 -55 443 141 -236 -199 -474 -115 -14 379 24 -196 236 475 24 496 -34 470 291 -328 231 119 418 -141 249 -315 -135 -457 39 -313 -59 -151 484 236 490 100 -111 -132 486 375 99 362 -321 -164 337 203 -168 -196 -474 -25 328 257 -405 -402 -32 -304 -216 -315 -261 323 -127 -468 24 -143 -380 367 458 9 -265 -204 236 -313 -342 415 23 -152 470 355 152 496 183 332 -395 -370 430 425 -174 -286 -389 417 -110 -16 -199 -234 -307 -79 -367 3 -218 221 299 -482 408 -190 433 -69 158 -245 138 162 -248 173 -6 209 303 -76 135 -19 491 -402 -102 233 82 199 499 -373 -28 -15 131 254 58 282 272 -34 -56 -443 -251 -46 -187 -261 -384 417 -235 110 126 -432 -113 113 50 230 211 -200 315 145 -148 -334 -375 324 151 256 79 61 390 203 -121 -165 261 480 289 -426 72 -242 491 337 368 469 405 107 435 307 189 146 -392 4 144 -40 171 269 136 174 377 -285 88 267 419 467 -46 180 448 -256 -394 20 2 -51 -143 222 -82 114 182 205 -78 -129 -148 30 -272 496 -158 -249 -383 478 425 494 46 365 261 317 333 -284 349 -367 312 455 153 166 -96 -138 -112 -326 476 -78 -268 250 -354 84 132 -126 432 474 125 -451 -195 402 -105 351 268 -344 168 453 224 17 -414 36 324 -409 -298 80 -47 443 -394 281 -135 -162 384 
500 -52137
-226 16 -263 58 343 -286 459 -352 -31 -146 -149 -263 363 -129 -310 87 -260 128 -376 -436 219 178 496 24 473 -398 -195 191 293 41 54 -81 -442 292 478 401 -142 437 -99 327 144 252 -84 7 -377 458 -406 -137 86 70 279 157 101 275 -319 74 -271 339 -235 -126 -120 -328 294 -62 -184 272 191 174 -439 -408 354 205 344 -230 64 -181 -271 159 182 -333 81 313 325 -318 88 -494 -391 -330 -303 226 44 78 398 190 -132 -286 314 -440 389 -272 4 -405 433 201 -135 350 -480 446 9 -445 -386 -58 368 -209 477 309 149 -414 -169 347 312 -273 277 -437 418 -355 -223 232 57 18 460 -86 113 -254 -385 -169 -404 -365 -223 457 42 243 251 263 -114 228 424 -464 -334 -245 -265 479 -18 -488 -106 -100 9 171 -15 -433 42 297 -19 -345 -457 448 338 -9 -65 -384 300 478 -289 52 -407 98 -220 -483 134 -53 -228 369 278 254 -267 24 7 242 -305 -156 -339 -411 -359 494 97 -315 442 -65 -472 230 51 329 208 -385 233 301 213 -135 -330 199 -336 -58 68 -206 48 -347 318 -93 -253 -486 251 -91 -397 393 255 -448 -70 198 -12 -42 280 -109 139 -12 6 224 141 -281 -410 -337 418 -246 105 338 49 5 491 219 413 -261 -415 -484 -500 41 409 255 93 191 305 -419 2 85 -27 -359 -75 479 366 -82 -301 -192 81 -31 62 38 -192 -37 -456 151 -317 -191 -258 -232 -175 -258 309 87 -150 255 -222 7 336 280 -407 -191 -226 370 -211 -8 -211 340 300 370 161 214 -91 -179 30 305 473 -435 -386 67 185 291 -338 347 230 364 102 -139 -129 -62 -7 316 -400 267 187 241 111 -172 81 263 50 94 330 -189 416 212 -384 -259 277 82 308 314 226 -178 161 -44 -314 115 317 58 -94 -189 -126 6 -70 -87 -253 -106 241 -320 -343 -208 274 -161 103 -458 -449 -428 -217 -320 154 -56 495 232 266 -492 -311 453 124 -142 363 -470 -479 89 388 304 -497 -365 -302 -404 -185 207 240 -59 -453 -156 -16 -402 268 267 -221 274 63 126 7 -170 134 -452 135 -242 -94 -2 140 280 -61 28 84 -58 -485 282 39 182 -11 131 -376 388 475 460 487 -405 -421 118 370 143 -256 -123 473 378 277 460 -11 35 310 -19 -185 -251 -490 -249 192 -123 -115 83 -440 375 -286 -316 263 42 -356 -398 489 -277 220 359 366 464 88 -309 343 -135 3 184 -247 -187 -335 68 415 27 -328 459 405 57 42 465 284 -392 1 400 -498 145 -498 492 -280 223 -297 -61 -461 -356 
500 41982
-266 9 -14 418 -386 -349 -64 -465 -434 463 -293 -475 368 264 419 333 49 -120 -166 -51 234 -169 -49 226 404 26 -218 -157 418 -74 -175 4 -213 -337 -77 -98 167 -289 -63 233 -326 144 -389 -105 260 -118 80 -191 262 415 -390 -3 98 62 75 2 -60 -291 -303 210 135 375 -285 -225 -110 490 177 -443 -299 -34 -357 227 -38 -246 -378 -278 488 203 384 251 470 494 100 68 408 27 -77 201 237 472 -89 -128 347 478 147 -262 468 176 147 21 -358 290 -251 456 -104 -129 -321 -115 -426 -437 -12 -104 -443 -412 -183 -182 467 240 -481 204 -436 -70 -423 412 -91 -424 2 229 -395 1 -249 -253 144 0 -296 -108 -277 235 -371 150 -350 117 -454 59 -443 -137 377 -475 -45 396 229 20 179 -342 284 -60 87 138 -331 192 -361 272 -209 -365 -228 -5 28 348 82 157 -2 232 -225 44 -208 184 -240 -331 209 215 -82 291 87 -51 -199 -129 389 388 -491 -90 80 1 -317 372 136 307 219 -336 155 -198 -326 5 -114 -51 -98 -470 -15 162 -300 47 -123 470 -162 -183 -81 139 188 160 380 -450 -430 -40 403 105 184 39 -87 256 -444 -80 -90 -270 426 296 -469 -172 327 16 490 -121 -437 219 201 253 -464 -380 245 -423 -368 125 -373 -298 -63 -470 308 -26 421 73 -270 -171 493 140 59 419 288 442 99 115 311 441 -154 -274 161 -453 480 -451 19 -423 -374 -497 202 105 -294 -9 135 -134 465 409 439 -305 -262 -216 335 -350 56 476 -408 -345 91 -245 -403 -210 -18 -390 189 -38 -341 61 391 -214 64 93 391 -378 -416 379 -12 -450 288 279 97 -122 -436 285 28 472 261 473 -373 204 -272 76 494 210 186 184 24 -302 -255 -85 336 -339 -140 -421 -216 297 -190 -376 -301 -50 404 296 329 -180 81 -143 -208 -158 182 271 -453 411 -152 -107 473 -114 -423 -2 84 174 -235 272 -312 126 352 -28 -77 -338 96 122 113 0 270 -58 -328 -148 -349 -36 -454 -314 -264 -407 -403 436 -13 -430 174 416 420 -241 443 38 -469 -369 -336 -265 103 439 -102 51 413 -489 -96 183 305 76 -113 -192 -107 -66 -6 481 379 443 417 -134 -134 91 135 -214 -298 78 324 86 209 -160 321 164 131 71 -285 44 82 119 80 239 48 -181 -100 441 253 394 422 485 338 -309 203 -296 -366 -162 342 -163 -232 -481 423 -171 -141 -404 493 -157 168 61 -261 102 32 -329 -158 -420 -9 -406 -127 96 488 -205 81 -322 338 -215 234 473 475 77 -190 -256 448 85 73 -193 -467 419 
500 238002
201 -20 -258 156 -136 -87 -150 297 404 -56 -330 353 -68 318 -66 111 156 -429 197 129 -453 -226 -209 -209 -278 376 364 382 -238 283 -264 463 115 -22 119 480 392 -179 277 -352 265 -201 -147 -302 -383 -360 -339 274 -289 358 255 -242 -15 -453 401 -441 -225 -382 -207 37 -247 30 353 369 -492 324 -299 -248 -354 -170 253 411 129 -394 -39 -401 -254 474 -275 310 -315 480 -80 -478 -121 -178 -67 -493 292 227 44 -455 109 249 -234 469 426 -33 74 -76 -351 -173 187 -369 285 -351 -270 -116 -25 307 46 -488 -361 466 -466 19 140 320 378 -68 47 274 330 8 24 -404 -23 302 416 -449 -274 -435 230 413 48 16 -86 278 -248 390 437 298 -246 -423 -384 141 448 257 -39 326 41 -140 100 -129 368 -24 320 -155 -222 236 -103 4 -347 -373 270 202 495 -316 332 99 -74 270 -103 181 -301 14 -178 -1 123 -365 -175 -336 -5 -223 388 -137 254 -440 208 -116 -352 -43 389 -199 437 -489 -145 -216 -305 40 384 474 162 -219 155 361 -353 -171 -140 -378 316 -463 139 -189 -186 379 -474 -80 439 -414 305 -413 44 -454 240 333 -443 -52 -383 104 488 -499 78 150 135 85 363 282 414 -277 -243 230 112 396 -107 426 275 -81 199 214 6 4 -347 -98 402 393 -265 459 -159 -148 415 -171 206 -6 -169 -159 -421 -306 -25 -154 269 232 -72 -119 128 322 160 -97 93 -141 -31 -401 -285 122 353 -383 368 -412 -72 -291 293 -157 391 -1 337 74 340 417 269 -185 115 38 -452 -457 -228 -472 -283 432 284 311 -209 -247 -238 6 -272 -384 475 -404 56 -97 157 -151 246 400 -300 84 -25 -108 353 96 208 -32 -366 -392 11 258 -12 229 190 -228 392 481 378 -346 339 -394 -230 314 -298 -321 -283 -289 28 -184 -388 -419 400 439 -27 -395 35 33 73 -479 493 -64 280 -18 17 -30 -394 -91 304 484 64 495 -410 -314 -338 144 -135 231 356 -254 -453 -180 179 447 -241 -348 -96 294 -462 477 -333 -469 266 447 365 -217 270 472 -455 426 456 461 -227 -453 -1 -65 -457 -283 -333 399 315 -286 -281 -6 14 330 146 -230 124 184 100 -356 -432 -134 -57 433 -499 65 405 -102 -9 -286 -141 265 113 359 52 156 -72 219 408 -257 -214 -373 237 -348 458 235 -78 434 272 -478 -70 340 240 374 -375 242 -61 -117 -360 -217 97 352 -452 62 -437 452 -282 -9 -476 -22 234 -190 106 -29 314 416 58 88 -298 -170 111 133 22 203 359 148 -55 -202 -469 438 81 480 -358 481 
500 81542
-295 434 -388 -452 458 91 282 120 49 105 -66 317 -337 -126 -129 346 485 -144 -132 189 -285 -484 -14 366 -101 276 299 -121 -230 281 421 -25 -433 -114 23 -123 -171 157 -151 378 114 283 -453 -370 9 -82 476 -5 275 -156 36 -158 213 -126 -292 -388 3 -492 -8 273 -359 265 -251 60 3 -376 437 332 -366 -214 -438 -400 -79 -391 -270 430 28 -294 277 155 403 165 497 468 -460 58 -420 43 66 424 -184 207 -458 417 -233 45 -106 56 -122 380 342 292 480 115 402 -289 397 282 269 175 289 -476 -160 286 492 -120 -156 -427 -225 -238 -151 444 -179 -109 -287 440 437 107 -4 167 -13 190 -41 320 157 213 -117 55 -5 4 -270 -216 29 422 -77 21 -197 -381 446 430 -118 -204 374 -445 -461 -412 496 476 -453 344 143 387 35 -45 207 44 -332 -58 -401 -484 446 181 -200 -25 104 -425 -151 -241 47 147 -311 -219 -205 -84 -164 -165 356 184 -189 403 29 -193 -358 -84 -386 -299 460 -218 143 -440 -202 90 241 451 -83 -155 -122 -382 456 425 266 -354 -294 -87 414 43 248 270 -273 -88 -475 -392 -429 -332 24 -315 221 -15 319 -135 397 -382 -193 490 -79 -424 188 299 -305 144 225 -187 -358 -217 -422 -444 326 327 326 -94 91 -296 14 162 -128 -109 -301 -407 -272 -482 310 -23 -12 -31 319 409 46 7 -439 -407 4 286 258 146 -431 336 -445 -252 -485 -267 -494 106 -63 372 120 309 -237 319 -245 343 -310 -83 -180 178 387 -360 440 285 499 -499 -270 3 -361 488 2 60 -324 409 -340 -308 -358 -334 -350 80 -461 271 241 154 442 -152 -2 -368 266 -330 -337 -495 -190 -45 290 -190 308 -128 -335 -201 212 -481 359 388 428 -480 -68 -77 38 83 -497 429 -146 -404 84 -204 -55 -66 281 -289 104 -204 68 415 251 210 77 411 -418 -406 -290 294 -34 -79 -466 -106 293 -33 -331 -168 -98 -328 -239 256 -231 345 -96 -434 131 37 -223 -264 -315 -303 3 -212 407 -68 -301 -159 26 -91 -13 492 -317 21 239 476 -160 408 -192 242 -67 70 350 -446 267 107 -380 -101 -356 249 487 -170 446 -10 118 -295 422 -330 46 300 431 -467 -355 -386 406 384 -409 -253 -356 -101 -159 77 -179 -308 483 -59 299 -397 -308 -205 352 -321 477 -350 -479 448 -145 -205 -382 401 -405 -451 434 -260 16 341 -24 107 -60 121 -142 281 50 -468 473 34 -27 -376 137 -483 -80 342 -304 249 492 69 197 -152 364 -333 -251 459 69 36 52 -415 229 -472 192 -331 -499 -98 -198 52 
