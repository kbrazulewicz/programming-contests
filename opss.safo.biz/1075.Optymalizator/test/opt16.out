NIE
TAK
NIE
TAK
NIE
TAK
TAK
TAK
NIE
NIE
